api/open_saves.proto

```diff
@@ -89,6 +89,76 @@ service OpenSaves {
   // The string is optional and the server returns an empty string if
   // omitted.
   rpc Ping(PingRequest) returns (PingResponse) {}
+
+  // CompareAndSwapProperty compares the property to old_value and updates the property to value 
+  // if the old_value and the current property are equal.
+  // The updated field in CASPropertyResponse is set to true if the swap is executed.
+  // For example, CompareAndSwapProperty(property, value = 42, old_value = 24) will set the
+  // property to 42 if the current value is 24.
+  // CompareAndSwapProperty also supports swapping with a value of another type, e.g. 
+  // CompareAndSwapProperty(property, value = "42", old_value = 24).
+  // Otherwise it will not update the property and return the current (unchanged) value
+  // and updated = false. The operation is executed atomically.
+  // Errors:
+  //   - NotFound: the requested record or property was not found.
+  rpc CompareAndSwapProperty(CASPropertyRequest) returns (CASPropertyResponse) {}
```

Comment:
I prefer the less cryptic function and argument names as such:
`rpc CompareAndSwap(CompareAndSwapRequest) returns (CompareAndSwapResponse) {}`

When function name or parameters are not using consistent naming, it makes it hard to remember.